src/CommonProviderImplementation/Helpers.fs

```diff
@@ -302,10 +304,10 @@ module internal ProviderHelpers =
                         providedTypesCache.TryRemove key |> ignore
                     else
                         log (sprintf "Saving generation of type %s for 10 seconds awaiting incremental recreation [%d]" fullTypeName tp.Id)
-                        providedTypesCache.[key] <- CacheValue (providedType, fullKey)
+                        providedTypesCache.AddOrUpdate(key, CacheValue (providedType, fullKey), fun _ _ -> CacheValue (providedType, fullKey)) |> ignore
```

Comment:
Given that the new value being added is not dependent on the old value, there is no gain in using the more complex AddOrUpdate overload